[Kevin Limantoro] iP

kevinlimantoro123 opened the pull request:
# HelperBot

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/project.html).
2. double click it
3. add your tasks
4. let it manage tasks for you 😉

And it is **FREE!**

Features:

- [X]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

___
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Rename main class an added greetings

Commit:
Add support for echo

Commit:
Add support for list

Commit:
Added mark as done feature and new Task class

Commit:
Created Todo, Event, and Deadline

Commit:
Added test cases

Commit:
Added error handling

Commit:
Added error handling

Commit:
Added delete function

Commit:
Fixed bugs and incorporated enums

Commit:
Standardize code with minor changes

Commit:
Implement save feature

Commit:
Minor changes

Commit:
Merge branch 'branch-Level-7'

Commit:
Add date and time function

Commit:
Test

kevinlimantoro123 committed:
Merge pull request #2 from kevinlimantoro123/branch-Level-8

Add date and time function

kevinlimantoro123 committed:
Merge branch 'master' into branch-Level-7

kevinlimantoro123 committed:
Merge pull request #1 from kevinlimantoro123/branch-Level-7

Test

Commit:
Refactor code to appropriate classes

Commit:
Reorganize files into packages

Commit:
Change minor details

kevinlimantoro123 committed:
Merge pull request #3 from kevinlimantoro123/add-gradle-support

Add gradle support

Commit:
Configure gradle to run program

Commit:
Create JUnit tests for MarkCommand, UnmarkCommand and Todo

Commit:
Add Java Doc to all methods, classes and interfaces

Commit:
Adjust code to follow coding standard

Commit:
Implement find function

kevinlimantoro123 committed:
Merge pull request #4 from kevinlimantoro123/A-JavaDoc

A-JavaDoc

kevinlimantoro123 committed:
Merge pull request #5 from kevinlimantoro123/A-CodingStandard

Adjust code to follow coding standard

kevinlimantoro123 committed:
Merge pull request #6 from kevinlimantoro123/branch-Level-9

Implement find function

Commit:
Create simple GUI

Commit:
Add more GUI functions

Commit:
Add GUI

Commit:
Add and fix checkstyle

kevinlimantoro123 committed:
Merge pull request #7 from kevinlimantoro123/branch-Level-10

Branch level 10

Commit:
Add assertions to key points in code

kevinlimantoro123 committed:
Merge pull request #8 from kevinlimantoro123/branch-A-Assertions

Add assertions to key points in code

Commit:
Refactor code to improve code quality

kevinlimantoro123 committed:
Merge pull request #10 from kevinlimantoro123/master

merge main to branch-A-CodeQuality

kevinlimantoro123 committed:
Merge pull request #11 from kevinlimantoro123/branch-A-CodeQuality

Branch a code quality

Commit:
Remove unecessary code

Commit:
Add priority tagging feature

kevinlimantoro123 committed:
Merge pull request #12 from kevinlimantoro123/branch-C-Priority

Add priority tagging feature

Commit:
Update relevant tests

SherrinfordHR commented on src/main/java/helperbot/parser/Parser.java:
```diff
@@ -0,0 +1,88 @@
+    }
+    private static Task createTask(char type, String description, int priority) {
+        switch (type) {
+        case 'T':
```
Good to see no indentation for case clauses.

SherrinfordHR commented on src/main/java/helperbot/command/AddCommand.java:
```diff
@@ -0,0 +1,104 @@
+        return "Got it. I've added this task:\n" + newTask + "\nNow you have "
+                + taskList.size() + (taskList.size() > 1 ? " tasks in the list." : " task in the list");
+    }
+    private Task createTask(String taskType, String description, int priority) {
```
Perhaps JavaDoc comments could be added?

SherrinfordHR commented on src/main/java/helperbot/HelperBot.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    /**
+     * Gets the response based on the user input.
```
Very clear JavaDoc comments throughout

SherrinfordHR commented on src/main/java/helperbot/HelperBot.java:
```diff
@@ -0,0 +1,68 @@
+/**
+ * The main class of the HelperBot program.
+ */
+public class HelperBot {
```
Generally good coding standard with clear JavaDoc comments and easy to understand variable and method name

PriyadarshiCharvi commented on src/main/java/helperbot/command/AddCommand.java:
```diff
@@ -0,0 +1,104 @@
+     *
+     * @param storage The storage handler.
+     * @throws IOException If an I/O error occurs.
+     */
```
Great job on the consistent javadocs! Very clean and readable.

PriyadarshiCharvi commented on src/main/java/helperbot/command/Command.java:
```diff
@@ -0,0 +1,12 @@
+package helperbot.command;
+
+import java.io.IOException;
+
+import helperbot.task.Storage;
+import helperbot.task.TaskList;
```
I like your consistent organisation of import statements.

PriyadarshiCharvi commented on src/main/java/helperbot/gui/Main.java:
```diff
@@ -0,0 +1,34 @@
+
+import java.io.IOException;
+
+import helperbot.HelperBot;
+import javafx.application.Application;
```
Could you consider separating different imports for consistency and easier readability?

PriyadarshiCharvi commented on src/main/java/helperbot/parser/Parser.java:
```diff
@@ -0,0 +1,88 @@
+        }
+    }
+    private static String extractDescription(String task, boolean isDone) {
+        String[] parts = task.split("\\(Priority: ");
+        String descriptionPart = parts[0];
+        System.out.println(descriptionPart);
```
Good job on following the correct naming standards, for example, plural for collections, camelCase, etc

PriyadarshiCharvi reviewed (commented):
Overall, clean and readable code. Good job on following the coding standards. Great work!

kevinlimantoro123 committed:
All classes: No assertions in code

Assertions allow to test the correctness of any assumptions that is made in the program (eg. index cannot be null)

Let's implement assertions in relevant parts of the code

Code changes were already done last week

kevinlimantoro123 committed:
All classes: No assertions in code

Assertions allow to test the correctness of any assumptions that is made in the program (e.g. index cannot be null)

Let's implement assertions in relevant parts of the code

Code changes were already done last week

kevinlimantoro123 committed:
Command and Parser classes: Code is too long and unclear

A clear structure is needed for easier debugging and better understanding

Refactor code to abstract similar functions and make code clearer

Let's group common behaviors to command and parser classes functions into new separate functions

Code changes were already done last week

kevinlimantoro123 committed:
Command and Parser classes: Code is too long and unclear

A clear structure is needed for easier debugging and better understanding

Refactor code to abstract similar functions and make code clearer

Let's group common behaviors to command and parser classes functions into new separate functions

Code changes were already done last week

kevinlimantoro123 committed:
Add more testing

kevinlimantoro123 committed:
Fix find command bugs

kevinlimantoro123 committed:
Add date/time to event tasks

kevinlimantoro123 committed:
Add user guide for HelperBot

kevinlimantoro123 committed:
Fix readme formatting

kevinlimantoro123 committed:
Tidy up code

kevinlimantoro123 committed:
Fix issues on missing data file

kevinlimantoro123 committed:
Edit readme docs

kevinlimantoro123 committed:
Edit minor changes to readme

kevinlimantoro123 committed:
Fix minor bugs in code

kevinlimantoro123 committed:
Fix code format

kevinlimantoro123 committed:
Edit code to follow standard

kevinlimantoro123 committed:
Move checkstyle files
